modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -771,10 +782,25 @@ Mat getMatFromTensor(const opencv_onnx::TensorProto& tensor_proto)
     {
         // FIXME, for now, we only load FP16 Tensor as FP32 Mat, full support for FP16 is required in the future.
         CV_LOG_ONCE_WARNING(NULL, "DNN: load FP16 model as FP32 model, and it takes twice the FP16 RAM requirement.");
-        if (!tensor_proto.float_data().empty())
+
+        // ONNX saves float 16 data in two format: int32 and raw_data.
+        // Link: https://github.com/onnx/onnx/issues/4460#issuecomment-1224373746
+        if (!tensor_proto.int32_data().empty())
         {
-            const ::google::protobuf::RepeatedField<float> field = tensor_proto.float_data();
-            Mat(sizes, CV_32FC1, (void*)field.data()).copyTo(blob);
+            const int offset = isLittleEndianCPU() ? 0 : 1;
```

Comment:
IIRC, we have this check in CMake scripts. Can we use a precomp macro here?

Reply:
Please reuse `WORDS_BIGENDIAN` macro (from cvconfig.h)